test/LockProxyTest.js

```diff
@@ -0,0 +1,180 @@
+    });
+
+    afterEach(function () {
+        return Q.all([lockOne.destroy(), lockTwo.destroy()]);
+    });
+
+    after(function () {
+        clientOne.shutdown();
+        return Controller.shutdownCluster(cluster.id);
+    });
+
+    it("locks and unlocks", function () {
+        this.timeout(10000);
+        var startTime = Date.now();
+        return lockOne.lock().then(function () {
+            setTimeout(function () {
+                lockOne.unlock();
+            }, 1000);
+            return lockTwo.lock()
+        }).then(function () {
+            var elapsed = Date.now() - startTime;
+            expect(elapsed).to.be.greaterThan(1000);
+        });
+    });
+
+    it("unlocks after lease expired", function () {
+        this.timeout(10000);
+        var startTime = Date.now();
+        return lockOne.lock(1000).then(function () {
+            return lockTwo.lock();
+        }).then(function () {
+            var elapsed = Date.now() - startTime;
+            expect(elapsed).to.be.greaterThan(1000);
+        });
+    });
+
+    it("gives up attempt to lock after timeout is exceeded", function () {
+        this.timeout(10000);
+        return lockOne.lock().then(function () {
+            return lockTwo.tryLock(1000);
+        }).then(function (acquired) {
+            expect(acquired).to.be.false;
+        });
+    });
+
+    it("acquires lock before timeout is exceeded", function () {
+        this.timeout(10000);
+        var startTime = Date.now();
+        lockOne.lock(1000);
+
+        return lockTwo.tryLock(2000).then(function (acquired) {
+            var elasped = Date.now() - startTime;
+            expect(acquired).to.be.true;
+            expect(elasped).to.be.greaterThan(1000);
+        });
+    });
+
+    it("acquires the lock before timeout and unlocks after lease expired", function () {
+        this.timeout(10000);
+        var startTime = Date.now();
+        return lockOne.lock(1000).then(function () {
+            return lockTwo.tryLock(2000, 1000);
+        }).then(function () {
+            var elapsed = Date.now() - startTime;
+            expect(elapsed).to.be.greaterThan(1000);
+            return lockOne.lock(2000);
+        }).then(function () {
+            var elapsed = Date.now() - startTime;
+            expect(elapsed).to.be.greaterThan(1000);
+        });
+
+    });
+
+    it("correctly reports lock status when unlocked", function () {
+        return lockOne.isLocked().then(function (locked) {
+            expect(locked).to.be.false;
+        });
+    });
+
+
+    it("correctly reports lock status when locked", function () {
+        return lockOne.lock().then(function () {
+            return lockOne.isLocked();
+        }).then(function (locked) {
+            expect(locked).to.be.true;
+            return lockTwo.isLocked();
+        }).then(function (locked) {
+            expect(locked).to.be.true;
+        });
+    });
+
+    it("correctly reports remaining lease time", function () {
```

Comment:
Is this test reliable? What if more than 100 milliseconds pass before `getRemainingLeaseTime` returns? This may cause irrelevant failures in PR builder in future.

Maybe we can call `getRemainingLeaseTime` two times one after the other. Then assert that first response is greater than the second.